src/coreComponents/meshUtilities/CPMesh/CPMeshParser.cpp

```diff
@@ -0,0 +1,372 @@
+  localIndex const nZ = cPMeshDims.nZ();
+
+  localIndex const maxSize = nX*nY*nZ; // TODO: change
+  std::vector< real64 > work;
+  work.reserve( maxSize );
+
+  string buffer = extractDataBelowKeyword( meshFile ); // TODO: change to save only what is necessary
+  CPMeshStringUtilities::eclipseDataBufferToVector( buffer, work ); // TODO: change to save only what is necessary
+  std::replace( work.begin(), work.end(), 2, 0 );
+  std::replace( work.begin(), work.end(), 3, 0 );
+
+  localIndex const nXLocal = cPMeshDims.nXLocal();
+  localIndex const nYLocal = cPMeshDims.nYLocal();
+  localIndex const nZLocal = cPMeshDims.nZLocal();
+
+  localIndex const iMinLocal = cPMeshDims.iMinLocal();
+  localIndex const jMinLocal = cPMeshDims.jMinLocal();
+  localIndex const kMinLocal = 0;
+
+  m_actnum.resize( nXLocal*nYLocal*nZLocal );
+
+  // Note: Below I mimic what I will ultimately do: not storing the full file in a string,
+  //       but instead going through the file and saving only what the MPI rank needs
+  for( localIndex k = 0; k < nZLocal; ++k )
+  {
+    for( localIndex j = 0; j < nYLocal; ++j )
+    {
+      for( localIndex i = 0; i < nXLocal; ++i )
+      {
+        localIndex const eiLocal = k*nXLocal*nYLocal + j*nXLocal + i;
+        localIndex const ei = (k+kMinLocal)*nX*nY + (j+jMinLocal)*nX + i+iMinLocal;
+        m_actnum( eiLocal ) = work[ ei ];
+      }
+    }
+  }
+}
+
+void CPMeshParser::readLocalPROP( std::istringstream & meshFile,
+                                  CPMeshDimensions const & cPMeshDims,
+                                  array1d< real64 > & prop )
+{
+  localIndex const nX = cPMeshDims.nX();
+  localIndex const nY = cPMeshDims.nY();
+  localIndex const nZ = cPMeshDims.nZ();
+
+  localIndex const maxSize = nX*nY*nZ; // TODO: change
+  std::vector< real64 > work;
+  work.reserve( maxSize );
+
+  string buffer = extractDataBelowKeyword( meshFile ); // TODO: change to save only what is necessary
+  CPMeshStringUtilities::eclipseDataBufferToVector( buffer, work ); // TODO: change to save only what is necessary
+
+  localIndex const nXLocal = cPMeshDims.nXLocal();
+  localIndex const nYLocal = cPMeshDims.nYLocal();
+  localIndex const nZLocal = cPMeshDims.nZLocal();
+
+  localIndex const iMinLocal = cPMeshDims.iMinLocal();
+  localIndex const jMinLocal = cPMeshDims.jMinLocal();
+  localIndex const kMinLocal = 0;
+
+  prop.resize( nXLocal*nYLocal*nZLocal );
+
+  // Note: Below I mimic what I will ultimately do: not storing the full file in a string,
+  //       but instead going through the file and saving only what the MPI rank needs
+
+  for( localIndex k = 0; k < nZLocal; ++k )
+  {
+    for( localIndex j = 0; j < nYLocal; ++j )
+    {
+      for( localIndex i = 0; i < nXLocal; ++i )
+      {
+        localIndex const eiLocal = k*nXLocal*nYLocal + j*nXLocal + i;
+        localIndex const ei = (k+kMinLocal)*nX*nY + (j+jMinLocal)*nX + i+iMinLocal;
+        prop( eiLocal ) = work[ ei ];
```

Comment:
These readers share a lot of codes, is it profitable to mix them up? For example, the reading of `actnum` and `prop` are almost identical.